src/samples.jl

```diff
@@ -117,5 +117,100 @@ function _sample_unit_nball_muller!(D::Vector{Vector{N}}, n::Int, p::Int;
     return D
 end
 
+## Rejection Sampling
+
+"""
+    _canonical_length(X::LazySet)
+
+Get the canonical length of the box-approximation of the convex set `X`.
+
+### Inputs
+
+- `X`   - lazyset
+
+### Outputs
+
+Matrix with two columns and `n=dims(X)` rows. Each column stands for
+one dimension of `X` whereas the first column is the minimum and the second
+column the maximum value of the corresponding dimension.
+"""
+function _canonical_length(X::LazySet{N}) where {N<:Real}
+    dims = dim(X)
+    x = Matrix{N}(undef, dims, 2)
+    for j=1:dims
+        ej = SingleEntryVector(j, dims, one(N))
+        x[j,:] = [-ρ(-ej, X), ρ(ej, X)]
+    end
+    return x
+end
+
+"""
+    rand(X::LazySet{N}, n_samples::Int;
+           rng::AbstractRNG=GLOBAL_RNG,
+           seed::Union{Int, Nothing}=nothing) where {N}
+
+Rejection sampling of an arbitrary LazySet `X` for which the support value
+function is defined.
+
+Draw a sample `x` from a uniform distribution of a box-overapproximation of the
+original set `X` in all `n` dimensions. The function rejects a drawn sample `x`
+and redraws as long as the sample is not contained in the original set `X`,
+i.e., `x ∉ X`.
+
+### Input
+
+- `X`           -- lazyset
+- `n_samples`    -- number of random samples
+- `rng`         -- (optional, default: `GLOBAL_RNG`) random number generator
+- `seed`        -- (optional, default: `nothing`) seed for reseeding
+
+### Output
+
+A vector of `n_samples` vectors.
+"""
+function Base.rand(X::LazySet{N}, n_samples::Int;
+                rng::AbstractRNG=GLOBAL_RNG,
+                seed::Union{Int, Nothing}=nothing) where {N<:Real}
+    rng = reseed(rng, seed)
+    @assert isbounded(X) "this function requires that the set `X` is bounded, but it is not"
```

Comment:
Huh, why should the set be bounded?

Reply:
Rejection sampling proceeds by taking a box overapproximation of the set, and sample inside that box. If the set is unbounded, the box will have infinities. So in that case we  should sample from a distribution with unbounded support. It's ok to restrict to bounded sets in this issue (and open a new issue for unbounded sets so we don't forget).

Reply:
I was referring to [this comment](https://github.com/JuliaReach/LazySets.jl/pull/1612#discussion_r314941390). The only reason why we kept the function `_canonical_length` was because it additionally supports unbounded sets. And this assertion makes that case break.

Note that `rand` is never tested for unbounded sets here (and I bet it would fail).